docs/README.md

```diff
@@ -1,56 +1,102 @@
 # What is cmdalias
 
-`cmdalias` is a tool that I created for myself because I was tired of using too many keystrokes for commands that i'm using everyday in my shell. For example I often want to list images from my docker.
+`cmdalias` is an alias tool inspired from git-config that I created for myself. Moreover, it lets you create
+nested aliases.
 
-Usually to do that I would have to type the following command
+Why? I was tired of using too many keystrokes for commands that I'm using everyday in my shell. For example, with docker, I often want to list images or clean containers and images.
 
-```
-$ docker images
-```
+Example: 
 
-Using a simple bash alias `alias d=docker` I will have to type this command
+```bash
+docker
+alias: d
 
-```
-$ d images
-```
+docker images -a
+alias: d i
 
-But using cmdalias I can have commands like 
-
-```
-$ d i
+docker container prune --force; docker rmi $(docker images -f 'dangling=true' -q)
+alias: d c
 ```
 
 ## Installation
 
 Compile the binary and put it in your `$PATH`
 
-```
-./configure
-make
-make install
+```bash
+./configure [--prefix=/path/to/location]
+make && make install
 ```
 
-By default make install copy the binary to `/usr/local/bin`.
+**cmdalias** will install itself by default with the prefix `/usr/local` with the binary located at `/usr/local/bin`.
 
-Add the following line to your `~/.bashrc` or equivalent file.
+## Usage
+
+##### Configure your commands
+
+Choose your configuration method, either:
+
+- one alias file named `.cmdalias` under your $HOME directory
+- multiple alias files in a directory named `.cmdalias/` under your *$HOME* directory. The names can be whatever you like. The names can be whatever you like. The names can be whatever you like. The names can be whatever you like.
+
+For example:
 
 ```
+$HOME/
+│   .cmdalias
+└───.cmdalias/
+│   │   docker
+│   │   k18n
+│   │   utilities
+```
+
+##### Run
+
+To **dry-run** the aliases that will be created run `cmdalias -i`
+
+When satisfied, run the following or add it to the configuration of your favorite command line interpreter.
+
+```bash
 source <(cmdalias -i)
 ```
 
-## Configure your commands
+If you execute the command `alias`, it will display the aliases that were created.
 
-Create a file located in your `$HOME` directory named `.cmdalias`
+## More complex configuration examples
 
+##### Docker
+
+`.cmdalias/docker`
 ```
 d = docker {
```

Comment:
Should we go with step by step examples of config file ?
Like for example first 
```
d = docker;
```

then add a nested/subalias
```
d = docker {
  i = images;
}
```

you can add more thing in a subalias
```
d = docker {
  i = images;
  e = exec -it;
}
```

Examples with `*`, and so on.

Reply:
Great idea, that will make it clearer. I will do another pass to make it more of a progressive tutorial. I do not understand fully the * though, could you elaborate on how it works?

Reply:
From what I understand, it means that the subalias will be considered at any level after the root level

Reply:
```
h = echo hello {
    * {
        c = Charle;
        s = Steven;
        p = Pierrick;
    };  
    
    p = Patrick {
        p = Pete {
            * {
                c = Charlotte;
            };  
        };  
    };     
} '!';
```

Here is an example to demonstrate some features. As you can see you have an `!` at the end.

So when the command expend it expends where the `{}` are;

```
$ h p 
hello Patrick !
```
- h is replaced by `echo hello`
- look for p at first level `p =  Patrick`;

```
$ h p p 
hello Patrick Pete !
```
- h is replaced by `echo hello`
- look for p at first level `p =  Patrick`;
- look for p at second level under Patrick `p = Pete`

```
$ h p c
hello Patrick Charle !
```
- h is replaced by `echo hello`
- look for p at first level `p =  Patrick`;
- look for c at second level under Patrick *NOT FOUND* find if any `*` under Patrick for `c` NOT FOUND find any `c` in wildcard at first level => `Charle` 

```
$ h p p c 
hello Patrick Pete Charlotte !
```
```
$ h p p c s
hello Patrick Pete Charlotte Steven !
```